misk/src/main/kotlin/misk/web/metadata/ServiceMetadataAction.kt

```diff
@@ -39,9 +39,8 @@ class ServiceMetadataAction @Inject constructor(
    * https://github.com/google/guice/wiki/FrequentlyAskedQuestions#how-can-i-inject-optional-parameters-into-a-constructor
    */
   @Singleton
-  class OptionalBinder @Inject constructor(@AppName val appName: String) {
+  class OptionalBinder @Inject constructor(@AppName val appName: String, deployment: Deployment) {
```

Comment:
While I was at it -- we can inject a deployment here, since web actions are wired up by guice. No need to get a new Deployment from an env check.